src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -109,7 +113,12 @@ def __init__(self):
 
     @classmethod
     def _register_osrandom_engine(cls):
-        _openssl_assert(cls.lib, cls.lib.ERR_peek_error() == 0)
+        # Clear any errors extant in the queue before we start. In many
+        # scenarios other things may be interacting with OpenSSL in the same
+        # process space and it has proven untenable to assume that they will
+        # reliably clear the error queue. Once we clear it here we will
+        # error on any subsequent unexpected item in the stack.
+        cls.lib.ERR_clear_error()
```

Comment:
Yeah, occasionally the error stack may not be empty when we start up and then we'll fail on future asserts where we expect to understand the state of our stack.